src/Compilers/CSharp/Portable/Binder/ContextualAttributeBinder.cs

```diff
@@ -64,5 +66,10 @@ internal Symbol AttributeTarget
                 return _attributeTarget;
             }
         }
+
+        protected override CSDiagnosticInfo SingleTypeNameNotFound(string whereText, DiagnosticBag diagnostics, Location location, AssemblySymbol forwardedToAssembly)
```

Comment:
I am taking back the statement that the implementation should be in ```Binder.LookupAttributeType```. It looks like might be able to reliably do this in Binder.NotFound based on LookupOptions.